src/v/storage/types.h

```diff
@@ -37,7 +37,7 @@ class snapshotable_stm {
     // offset already exists
     virtual ss::future<> ensure_snapshot_exists(model::offset) = 0;
     // hints stm_manager that now it's a good time to make a snapshot
-    virtual ss::future<> make_snapshot() = 0;
+    virtual void maybe_make_snapshot() = 0;
```

Comment:
`maybe_make_snapshot` name hints that sometimes it won't be made. Maybe use some other name like `make_snapshot_in_background`?

Reply:
good idea